src/components/TradeWidget/Price.tsx

```diff
@@ -156,13 +156,14 @@ export const PriceInputBox = styled.div<{ hidden?: boolean }>`
 `
 
 interface Props {
-  sellToken: TokenDetails
-  receiveToken: TokenDetails
+  quoteToken: TokenDetails
+  baseToken: TokenDetails
   priceInputId: string
   priceInverseInputId: string
   tabIndex?: number
   swapPrices: () => void
-  priceShown: 'INVERSE' | 'DIRECT'
+  isPriceInverted: boolean
+  wasPriorityAdjusted: boolean
```

Comment:
Why do we need the `wasPriorityAdjusted`  why is it relevant?
I mean, the important thing is what's the new quote, not if it's the one we thought originally, or not. Probably I'm missing sth

Reply:
BASE_TOKEN/QUOTE_TOKEN: `DAI/WETH`

`DAI` = BASE TOKEN
`WETH` = QUOTE TOKEN

`DAI` PRIORITY = 1
`WETH` PRIORITY = 3

SHOW MARKET AS: `WETH/DAI` i.e
`DAI` = QUOTE TOKEN
`WETH = BASE TOKEN

In this scenario, `PriceSuggestions` receive the `baseToken` and `quoteToken` props as:
(they were switched above, according to the priority scheme)
```
<PriceSuggestions
  ...
  baseToken={WETH}
  quoteToken={DAI}
  isPriceInverted={priceInvertedState}
/>

where, inside `PriceSuggestions` - the `usePriceWithSlippage` hook is run using the `baseToken` `quoteToken` specified props above

So the price using the adjusted tokens will return the `inversePrice` essentially as the token pair has been switched, meaning we need some way of knowing when it has been switched to adjust the pricing shown

Open to any easier approaches. One thing to keep in mind while testing is the user's choice to `inversePrice` and how to accurately portray that in the multiple places we show the inverse price button and the `Receive at least` + `Limit Price` inputs

Reply:
I understand why `isPriceInverted` is needed, but not `wasPriorityAdjusted`.

If you hand the final base/quote and the isPriceInverted, any component should be able to represent it.
Also, most components wouldn't even need the "isPriceInverted" probably. I can see why for price suggestions does.

> Open to any easier approaches

One idea is modeling as having JUST `sellToken`, `buyToken`,  `baseToken` and `quoteToken`.
the two first is what you have selected in the form
the two last define the market. Just thought this might model more what we have in there.